pkgs/development/libraries/criterion/default.nix

```diff
@@ -1,33 +1,41 @@
-{ lib, stdenv, fetchFromGitHub, boxfort, cmake, libcsptr, pkg-config, gettext
-, dyncall , nanomsg, python3Packages }:
+{ lib, stdenv, fetchFromGitHub, boxfort, meson, libcsptr, pkg-config, gettext
+, cmake, ninja, protobuf, libffi, libgit2, dyncall, nanomsg, nanopbMalloc
+, python3Packages, git }:
 
 stdenv.mkDerivation rec {
-  version = "2.3.3";
   pname = "criterion";
+  version = "2.4.1";
 
   src = fetchFromGitHub {
     owner = "Snaipe";
     repo = "Criterion";
     rev = "v${version}";
-    sha256 = "0y1ay8is54k3y82vagdy0jsa3nfkczpvnqfcjm5n9iarayaxaq8p";
+    sha256 = "KT1XvhT9t07/ubsqzrVUp4iKcpVc1Z+saGF4pm2RsgQ=";
     fetchSubmodules = true;
   };
 
-  nativeBuildInputs = [ cmake pkg-config ];
+  nativeBuildInputs = [ meson ninja cmake pkg-config protobuf ];
 
   buildInputs = [
     boxfort.dev
     dyncall
     gettext
     libcsptr
     nanomsg
+    nanopbMalloc
+    libgit2
+    libffi
   ];
 
   checkInputs = with python3Packages; [ cram ];
 
   cmakeFlags = [ "-DCTESTS=ON" ];
   doCheck = true;
-  checkTarget = "criterion_tests test";
+  checkTarget = "test";
+
+  preConfigure = ''
+    patchShebangs ci/isdir.py src/protocol/gen-pb.py
```

Comment:
is it required ? ci script shouldn't be called ? Ii tried with --pure in nix-shell and it wasn't needed.

Reply:
Removing it leads the configure script to fail with `ERROR: Could not execute command "/build/source/ci/isdir.py .git"`, even with `--pure`

Reply:
> is it required ? ci script shouldn't be called ? Ii tried with --pure in nix-shell and it wasn't needed.

It's just a silly 3 line script calling `os.path.isdir(sys.argv[1])`. No harm done.

Reply:
ok ok it is just that I build this software without needing this in my overlay.

Reply:
Well in my testing (pure, mind you) it doesn't work. Did you build from within the Nixpkgs tree or just the one file with 1 systemwide Nixpkgs channel?

Reply:
It is needed:

https://github.com/Snaipe/Criterion/blob/56f8f1a4d06eb144d1b7c6a26619ab6adff9fdc8/meson.build#L22

When I said no harm done, I meant that there is no need to patch `meson.build` to not run that file. I understand the concern that some file named `ci.py` may run some CI-only related stuff, but not in this case.